codelists/models.py

```diff
@@ -562,12 +563,60 @@ def table_with_fixed_headers(self):
         # Identify the index for the two columns we want
         code_header_index = header_row.index(code_header)
         term_header_index = header_row.index(term_header) if term_header else None
+
+        table_rows = self.table[1:]
+        if include_mapped_vmps and self.coding_system_id == "dmd":
+            # ignore include_mapped_vmps if coding system is anything other than dmd
+            term_header_index = term_header_index or len(self.table[0])
+            codes = [row[code_header_index] for row in table_rows]
+
+            # add in mapped VMP codes
+            vmp_to_prev_mapping, mapped_vmps_for_this_codelist = vmpprev_full_mappings(
+                codes
+            )
+            # vmp_to_prev_mapping is a simple 1:1 mapping of vmp ID to the immediate previous
+            # VMP that it replaced
+            # Since we're mapping later codes as well as previous ones, we also need the
+            # reverse mapping of previous VMP to the one that replaced it
+            prev_to_vmp_mapping = {v: k for k, v in vmp_to_prev_mapping.items()}
+
+            previous_vmps_to_add = set()
+            subsequent_vmps_to_add = set()
+
+            for vmp, previous_vmp in mapped_vmps_for_this_codelist:
+                if vmp in codes:
+                    # mapping a code in the codelist to a previous code
+                    # add the previous code to the list
+                    previous_vmps_to_add.add(previous_vmp)
+                else:
+                    assert previous_vmp in codes
+                    # mapping a code in the codelist to a new code that supercedes it
+                    subsequent_vmps_to_add.add(vmp)
+
+            assert not previous_vmps_to_add & subsequent_vmps_to_add
+
+            def add_row(vmp, description):
+                new_row = ["" for i in range(len(table_rows[0]) + 1)]
+                new_row[code_header_index] = vmp
+                new_row[term_header_index] = description
+                table_rows.append(new_row)
+
+            for vmp in previous_vmps_to_add:
+                # add the code to the table data
+                # include its description as the code it was superceded by
+                add_row(vmp, f"VMP previous to {prev_to_vmp_mapping[vmp]}")
```

Comment:
Is there any way we could include the name of the previous VMP?

Reply:
The tests made me think that we're not including the name -- eg "VMP previous to 10514511000001106"

Reply:
Oh, I see, you mean the term as well as the code? 
It depends; we'd have to look up the term in the right release of dm+d.  We only have the ability to do that for releases we've imported since the time when we introduced coding system releases into OpenCodelists.  There's not an easy way of telling which release the mapped code is from (we just know it's not in the release that this codelist was made with, and we know it is in one that's either older or newer, depending on whether the mapped code is a previous or future one), so we'd have to try them until we found one that contained it, which could be expensive.